rfc_backmatter.md

```diff
@@ -57,6 +57,26 @@ As another example, when encoding 32-bit PCM with fixed predictors, all predicto
 ## Rice coding
 When folding (i.e. zig-zag encoding) the residual sample values, no extra bits are needed when the absolute value of each residual sample is first stored in an unsigned data type of the size of the last step, then doubled and then has one subtracted depending on whether the residual sample was positive or negative. Many implementations however choose to require one extra bit of data type size so zig-zag encoding can happen in one step and without a cast instead of the procedure described in the previous sentence.
 
+# Past format changes
+
+This informational appendix documents what changes were made to the FLAC format over the years. This information might be of use when encountering FLAC files that were made with software following the format as it was specified initially.
+
+The FLAC format was first specified in December 2000 and the bitstream format was considered frozen with the release of FLAC 1.0 (the reference encoder/decoder) in July 2001. Only changes made since this first stable release are considered in this appendix. Changes made to the FLAC subset definition are not considered.
+
+## Addition of blocksize strategy flag
+
+Perhaps the largest backwards incompatible change to the specification was published in July 2007. Before this change, variable blocksize streams were not explicitly marked as such by a flag bit in the frame header. A decoder had two ways to detect a variable blocksize stream, either by comparing the minimum and maximum blocksize in the STREAMINFO metadata block (which are equal in case of a fixed blocksize stream), or, in case a STREAMINFO metadata block was not present in a file, by detecting a change of blocksize during a stream, which could in theory not happen at all. As the meaning of the coded number number in the frame header depends on whether or not a stream is variable blocksize, this presented a problem: the meaning of the coded number could not be reliably determined. To fix this problem, one of the reserved bits was changed to be used as a blocksize strategy flag. [See also the section frame header](#frame-header).
+
+Along with the addition of a new flag, the meaning of the [blocksize bits](#blocksize-bits) was subtly changed. Initially, blocksize bits 0b0001-0b0101 and 0b1000-0b1111 could only be used for fixed blocksize streams, while 0b0110 and 0b0111 could be used for both fixed blocksize and variable blocksize streams. With the change these restrictions were lifted and 0b0001-0b1111 are now used for both variable blocksize and fixed blocksize streams.
```

Comment:
> It would perhaps be better to recommend not to use variable blocksize streams at all if wide interoperability is a concern. That doesn't fit in this part though, so I think this concern (and the others suggested) are better moved to a wiki page and then linked from here and the section implementation status.

IMO and if I remember well IETF docs have sometimes a paragraph about interoperability.

It may be good to add another appendix at the end of this patch about interoperability issues e.g.

> # Wide interoperability
>
> At the time of writing of this document, several hardware FLAC decoders are already in the wild and can not be easily upgraded. Some of them don't support variable blocksize streams. Encoders willing to maintain interoperability with a wide range of hardware decoders are RECOMMENDED to avoid variable blocksize.`